collector/pipelinerun.go

```diff
@@ -0,0 +1,58 @@
+package collector
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	contextTimeout = 300 * time.Second
+)
+
+func SetupPipelineRunReconcilerWithManager(mgr ctrl.Manager, logger log.Logger) error {
+	//TODO if/when we move off of the stand alone prometheus integration, we wil create metrics using the
```

Comment:
Can you please elaborate here on what you mean by "standalone Prometheus integration?" Offhand I don't understand what this refers to.

Reply:
Sure I'll update the comment.

Preview:
- the controller Options struct used to create the manager has a metric bind address
- the "sigs.k8s.io/controller-runtime/pkg/metrics" package has a `Registry.MustRegister(statsCollector)` method that sets up the prometheus stack and wires your collector's Describe and Collect metrics to be called